src/components/story-element.js

```diff
@@ -94,12 +95,63 @@ const DEFAULT_TEMPLATES = {
 };
 
 class StoryElementBase extends React.Component {
+
+  constructor(props){
+    super(props);
+    this.observer = null;
+    this.storyElementRef = null;
+  }
+
+  componentDidMount() {
+    this.initiateObserver();
+  }
+
+  componentWillUnmount() {
+    this.destroyObserver();
+  }
+
+  initiateObserver = () => {
+    if(this.props.disableAnalytics === true) return false;
+
+    const options = {
+      root: null,
+      rootMargin: '0px',
+      threshold: 1.0
+    };
+    this.observer = new IntersectionObserver(this.observerCallback, options);
+    this.observer.observe(this.storyElementRef);
+  };
+
+  destroyObserver = () => {
+    this.observer && this.observer.disconnect();
+  };
+
+  observerCallback = entries => {
+    entries.forEach(entry => {
+      if (entry.isIntersecting) {
+        this.emitElementQlitics();
+        this.destroyObserver(); //destroy observer once the call is made
+      }
+    });
+  };
+
   template() {
     const storyElement = this.props.element;
     const templates = Object.assign({}, DEFAULT_TEMPLATES, this.props.templates);
     return templates[storyElement.subtype] || templates[storyElement.type] || "div";
   }
 
+
+  emitElementQlitics() {
+    const {story = {}, card = {}, element = {}} = this.props;
+    if(global.qlitics){
+      global.qlitics('track', 'story-element-view', getQliticsSchema(story,card,element));
+    } else {
+      global.qlitics=global.qlitics||function(){(qlitics.q=qlitics.q||[]).push(arguments);};
```

Comment:
can you explain what you are trying to do here?

Reply:
@gja recommended checking the earlier implementations, the code ripped from [this](https://github.com/quintype/quintype-node-framework/blob/0dbf683ea59547d5c7f71bfe9467a8c3c1762d3b/client/analytics.js#L6)

Reading through the code, If qlitics is not found, it gets allocated to a function which takes in arguments to push into qlitics.q